// @domino/domino-db/src/strings/errors.js
/* ©Copyright HCL Technologies Ltd. 2018 */
module.exports = {
  NAME_LOCAL_VALIDATION_FAILURE:
    'Invalid or no "filePath" provided. "filePath" must be a string.',
  UNID_LOCAL_VALIDATION: 'Invalid "unid" provided. "unid" must be a string.',
  /**
   * Server class errors
   */
  HOST_NAME_LOCAL_VALIDATION_FAILURE:
    'Invalid or no "hostName" provided. "hostName" must be a string.',
  CONNECTION_LOCAL_VALIDATION_FAILURE:
    'Invalid "connection" provided. "connection" must be an object.',
  CREDENTIALS_LOCAL_VALIDATION_FAILURE:
    'Credentials are required for a secure connection',
  CLIENT_KEY_LOCAL_VALIDATION_FAILURE:
    'clientKey is required for a secure connection',
  CLIENT_CERT_LOCAL_VALIDATION_FAILURE:
    'clientCertificate is required for a secure connection',
  /**
   * Database class errors
   */

  /**
   * View class errors
   */
  COLLECTION_UNID_OR_NAME_REQUIRED:
    'Either the view\'s "name" or "unid" must be specified.',

  /**
   * HTTP errors
   */
  UNEXPECTED_STATUS_CODE: 'Unexpected status code: ',
  RESPONSE_BODY_MALFORMED: 'Response body malformed: ',
  UNEXPECTED_RESPONSE_CONTENT_TYPE: 'Unexpected content type in response: ',

  /**
   * gRPC/Proton errors
   */
  GRPC_CLIENT_ERROR: 'gRPC client error',
  BULK_NOTE_REQUEST_FAILED: 'bulkNote request failed',
  BULK_NOTE_REQUEST_FAILED_WITH_CODE:
    'bulkNote request failed with Proton code ',
  BULK_ITEM_DELETE_NO_ITEMS:
    'Delete item request requires a list of item names',
  BULK_ITEM_REPLACE_NO_UNIDS: 'Replace item request requires a list of UNIDs',
  BULK_OPERATION_NO_QUERY: 'Bulk operation requires a query string',
  BAD_QUERY_ARGS_TYPE: 'queryArgs must be an array',
  BAD_QUERY_ARG_VALUE: 'Unknown query argument value type',

  /**
   * Local conversion errors
   */
  DOC_CONVERSION_UNKNOWN_ITEM_TYPE:
    'Cannot convert document. Unknown item type: ',
  DOC_CONVERSION_NUMBER_PAIR_LENGTH: 'Invalid length of number pair: ',
  DOC_CONVERSION_DATETIME_PAIR_LENGTH: 'Invalid length of datetime pair: ',
};
