// @domino/domino-db/src/requests/grpc/generated/domino_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

// Original file comments:
// **********************************************************************
//
//                                                                      
//
// Licensed Materials - Property of HCL                                 
//
//                                                                      
//
// (c)Copyright HCL Technologies Ltd. 2018. All Rights Reserved.        
//
//                                                                      
//
// Note to U.S.Government Users Restricted Rights :                     
//
// Use, duplication or disclosure restricted by GSA ADP Schedule        
//
// Contract with IBM Corp.                                              
//
//                                                                      
//
// **********************************************************************
//
'use strict';
var grpc = require('grpc');
var domino_pb = require('./domino_pb.js');

function serialize_domino_BulkNoteReq(arg) {
  if (!(arg instanceof domino_pb.BulkNoteReq)) {
    throw new Error('Expected argument of type domino.BulkNoteReq');
  }
  return new Buffer(arg.serializeBinary());
}

function deserialize_domino_BulkNoteReq(buffer_arg) {
  return domino_pb.BulkNoteReq.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_domino_BulkNoteRsp(arg) {
  if (!(arg instanceof domino_pb.BulkNoteRsp)) {
    throw new Error('Expected argument of type domino.BulkNoteRsp');
  }
  return new Buffer(arg.serializeBinary());
}

function deserialize_domino_BulkNoteRsp(buffer_arg) {
  return domino_pb.BulkNoteRsp.deserializeBinary(new Uint8Array(buffer_arg));
}


var ServerService = exports.ServerService = {
  //
  // rpc Init(InitReq) returns (InitRsp) {}
  //
  // rpc GetDatabase(GetDatabaseReq) returns (GetDatabaseRsp) {}
  //
  // rpc GetView(GetViewReq) returns (GetViewRsp) {}
  //
  // rpc GetViewEntries(GetViewEntriesReq) returns (GetViewEntriesRsp) {}
  //
  bulkNote: {
    path: '/domino.Server/BulkNote',
    requestStream: false,
    responseStream: false,
    requestType: domino_pb.BulkNoteReq,
    responseType: domino_pb.BulkNoteRsp,
    requestSerialize: serialize_domino_BulkNoteReq,
    requestDeserialize: deserialize_domino_BulkNoteReq,
    responseSerialize: serialize_domino_BulkNoteRsp,
    responseDeserialize: deserialize_domino_BulkNoteRsp,
  },
  bulkNoteStream: {
    path: '/domino.Server/BulkNoteStream',
    requestStream: true,
    responseStream: true,
    requestType: domino_pb.BulkNoteReq,
    responseType: domino_pb.BulkNoteRsp,
    requestSerialize: serialize_domino_BulkNoteReq,
    requestDeserialize: deserialize_domino_BulkNoteReq,
    responseSerialize: serialize_domino_BulkNoteRsp,
    responseDeserialize: deserialize_domino_BulkNoteRsp,
  },
};

exports.ServerClient = grpc.makeGenericClientConstructor(ServerService);
