// @domino/domino-db/src/requests/index.js
/* ©Copyright HCL Technologies Ltd. 2018 */
const grpcRequestsLib = require('./grpc');

/**
 * This export serves as the "libary" of calls that can be made
 *
 * As this Node.js module (and Domino) evolve, the method in which
 * calls are made may change. For example, both gRPC and HTTP calls
 * may be made to domino through this Node.js module.
 *
 * By using this export, we provide a single location to collect
 * all supported calls, regardless of what protocol is being used.
 */
module.exports.configure = ({ hostName, connection, credentials }) => {
  const grpcRequests = grpcRequestsLib.configure({
    hostName,
    connection,
    credentials,
  });

  return {
    server: {
      // No server requests for now
    },
    database: {
      createDocument: grpcRequests.database.createSingleDocument,
      explainQuery: grpcRequests.database.explainQuery,
      bulkCreateDocuments: grpcRequests.database.createDocuments,
      bulkReadDocuments: grpcRequests.database.readDocuments,
      bulkReadDocumentsByUnid: grpcRequests.database.readDocumentsByUnid,
      bulkDeleteDocuments: grpcRequests.database.deleteDocuments,
      bulkDeleteDocumentsByUnid: grpcRequests.database.deleteDocumentsByUnid,
      bulkDeleteItems: grpcRequests.database.deleteItems,
      bulkDeleteItemsByUnid: grpcRequests.database.deleteItemsByUnid,
      bulkReplaceDocumentsByUnid: grpcRequests.database.replaceDocumentsByUnid,
      bulkReplaceItems: grpcRequests.database.replaceItems,
      bulkReplaceItemsByUnid: grpcRequests.database.replaceItemsByUnid,
    },
    document: {
      read: grpcRequests.document.readSingleDocument,
      replaceItems: grpcRequests.document.replaceSingleDocumentItems,
      replace: grpcRequests.document.replaceSingleDocument,
      delete: grpcRequests.document.deleteSingleDocument,
      deleteItems: grpcRequests.document.deleteSingleDocumentItems,
    },
  };
};
