// @domino/domino-db/src/requests/http/index.js
/* ©Copyright HCL Technologies Ltd. 2018 */
const serverHttpRequests = require('./server.js');
const databaseHttpRequests = require('./database.js');
const documentHttpRequests = require('./document.js');
const { makeHttpUrl } = require('./utils/http-helpers.js');

module.exports.configure = ({ auth, hostName, protocols }) => {
  /*
  const requestPromiseLib = require('request-promise');
  const rp = requestPromiseLib.defaults({
    auth,
    baseUrl: makeHttpUrl({ hostName, protocols }),
    resolveWithFullResponse: true,
    simple: false,
  });
  */

  // An early prototype used request-promise to dispatch some
  // requests over HTTP. Since this isn't being used, we've
  // removed the request-promise dependency. To restore HTTP
  // request processing, replace the following line with the
  // code in the above block comment.
  const rp = { auth, baseUrl: makeHttpUrl({ hostName, protocols }) };

  return {
    server: serverHttpRequests(rp),
    database: databaseHttpRequests(rp),
    document: documentHttpRequests(rp),
  };
};
